plugins/modules/cohesity_storage_domain.py

```diff
@@ -19,9 +27,127 @@
 
 except Exception:
     pass
+
 cohesity_client = None
 TWENTY_GiB = 20 * (1024**3)
 EIGHTEEN_GiB = 18 * (1024**3)
+
+
+DOCUMENTATION = """
+---
+author: Naveena (@naveena-maplelabs)
+description:
+  - Ansible Module used to create or remove a storage domain from a Cohesity Cluster.
+  - When executed in a playbook the appropriate state action will be applied.
+module: cohesity_storage_domain
+options:
+  cluster:
+    aliases:
+      - cohesity_server
+    description:
+      - IP or FQDN for the Cohesity Cluster
+    type: str
+  cohesity_admin:
+    aliases:
+      - admin_name
+      - cohesity_user
+      - username
+    description:
+      - Username with which Ansible will connect to the Cohesity Cluster. Domain
+        Specific credentails can be configured in following formats
+      - AD.domain.com/username
+      - AD.domain.com/username@tenant
+      - LOCAL/username@tenant
+    type: str
+  cohesity_password:
+    aliases:
+      - password
+      - admin_pass
+    description:
+      - Password belonging to the selected Username. This parameter will not be
+        logged.
+    type: str
+  ad_domain_name:
+    description:
+      - Specifies the active directory name that the view box is mapped to.
+    required: false
+    type: str
+  cluster_partition_id:
+    default: 3
+    description:
+      - The Cluster Partition id where the Storage Domain (View Box) will be created.
+    type: int
+  state:
+    choices:
+      - present
+      - absent
+    default: present
+    description:
+      - Determines the state of the storage domain.
```

Comment:
Just mention present creates the storage domain and absent deletes the storage domain.